fluidBuild.config.cjs

```diff
@@ -206,6 +206,10 @@ module.exports = {
 				// TODO: AB#7630 uses lint only ts projects for coverage which don't have representative tsc scripts
 				"^packages/tools/fluid-runner/package.json",
 			],
+			"fluid-build-tasks-tsc": [
+				// False positive?
+				"^packages/test/test-end-to-end-tests/package.json",
```

Comment:
Without this exclusion, I get the pollowing policy errors:

```
file failed the "fluid-build-tasks-tsc" policy: packages/test/test-end-to-end-tests/package.json
'build:test:esm' task is missing the following dependency: 
        @fluid-tools/build-cli#build:test or @fluid-tools/build-cli#tsc
```

Very strange that it's looking for deps on build-cli? @jason-ha Do you know what's happening here?

Reply:
How do you connect your local build tools to client for testing?

Reply:
I usually don't, but when I do, I use pnpm.overrides. SOmething like:

```
"@fluid-tools/build-cli": "link:./build-tools/packages/build-cli",
"@fluidframework/build-tools": "link:./build-tools/packages/build-tools"
```

But the policy failure is happening in CI, too, so seems to be unrelated to my local setup.

Reply:
It is the change in recognized module type for build-cli that is tripping up the policy dependency checking.
1. We have a global task config that ^tsc is done for dependencies listed when running tsc.
2. A `tsc` script is now recognized for CommonJS or ESM. The policy roughly says is there is a dependency with a matching module type then the task dependency must be declared.
3. With your change build-cli `tsc` switched from CommonJS to ESM.

The dependency policy said the ESM of all of the packages that have build-cli listed must depend on build-cli `tsc` (or `build:test` since is doesn't know which one is the product build). The system does not have precise knowledge to distinguish library (import) dependencies versus execution dependencies. The post-build policy check I would like to see will be able to tell the difference.
So, for the approximate check we do now, I have added logic not to require dependencies across release groups. See #21238

Reply:
Thanks; I updated the comment with the link to your PR and noted that the override can be removed once client uses build-tools 0.39+.

Reply:
Hmm. Only one package needed override? When I was using locally built build-tools, I was getting reports for just about every package. If it is isolated, then client group isn't using the build-tools that restores deps checks for scripts that are not just `"tsc"`. That would mean instead of config suppression you could set build-cli "tsc" to `"tsc --project tsconfig.json"`, which is the common pattern we ended up using. I am good with either.
Going to unresolve just for visibility.

Reply:
I tried changing build-cli's tsc script to be `tsc --project tsconfig.json` but no change, so I left the config exclusion and comment.